backend/internal/db/dbgorm/deployments.go

```diff
@@ -24,27 +29,37 @@ func (db *gormdb) CreateDeployment(ctx context.Context, deployment sdktypes.Depl
 	}
 
 	if err := db.locked(func(db *gormdb) error {
-		return db.db.WithContext(ctx).Create(&e).Error
+		return db.createDeployment(ctx, d)
 	}); err != nil {
 		return translateError(err)
 	}
 	return nil
 }
 
+func (db *gormdb) getDeployment(ctx context.Context, deploymentID string) (*scheme.Deployment, error) {
+	return get1(db.db, ctx, scheme.Deployment{}, "deployment_id = ?", deploymentID)
+}
+
 func (db *gormdb) GetDeployment(ctx context.Context, id sdktypes.DeploymentID) (sdktypes.Deployment, error) {
-	return get(db.db, ctx, scheme.ParseDeployment, "deployment_id = ?", id.String())
+	if d, err := db.getDeployment(ctx, id.String()); d == nil {
+		return nil, err
+	} else {
+		return scheme.ParseDeployment(*d)
+	}
```

Comment:
```suggestion
	d, err := db.getDeployment(ctx, id.String()); 
	if err != nil || d == nil {
		return nil, err
	} 
	return scheme.ParseDeployment(*d)
```

Not a must, I personally, really don't like using else where I can, you can self resolve this one what ever you decide

specifically about the if condition, I think it's better to be verbose and explicit

Reply:
done. resolving